circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -132,44 +222,66 @@ private class CircuitSymbolProcessor(
           } else {
             FactoryType.UI
           }
+        val assistedParams =
+          fd.assistedParameters(symbols, logger, screenKSType, factoryType == FactoryType.PRESENTER)
         codeBlock =
           when (factoryType) {
             FactoryType.PRESENTER ->
               CodeBlock.of(
-                "%M·{·%M()·}",
+                "%M·{·%M(%L)·}",
                 MemberName("com.slack.circuit", "presenterOf"),
-                MemberName(packageName, name)
+                MemberName(packageName, name),
+                assistedParams
               )
             FactoryType.UI -> {
               val stateParam =
                 fd.parameters.singleOrNull { parameter ->
-                  parameter.type.resolve().isAssignableFrom(circuitUiState)
+                  symbols.circuitUiState.isAssignableFrom(parameter.type.resolve())
                 }
               if (stateParam == null) {
                 CodeBlock.of(
-                  "%M<%T>·{·%M()·}",
+                  "%M<%T>·{·%M(%L)·}",
                   MemberName("com.slack.circuit", "ui"),
                   CircuitUiState::class.java,
                   MemberName(packageName, name),
+                  assistedParams
                 )
               } else {
+                val block =
+                  if (assistedParams.isEmpty()) {
+                    CodeBlock.of("")
+                  } else {
+                    CodeBlock.of(",·%L", assistedParams)
+                  }
                 CodeBlock.of(
-                  "%M<%T>·{·state·->·%M(%L·=·state)·}",
+                  "%M<%T>·{·state·->·%M(%L·=·state%L)·}",
                   MemberName("com.slack.circuit", "ui"),
                   stateParam.type.resolve().toTypeName(),
                   MemberName(packageName, name),
-                  stateParam.name!!.getShortName()
+                  stateParam.name!!.getShortName(),
+                  block
                 )
               }
             }
           }
       }
       InstantiationType.CLASS -> {
         val cd = annotatedElement as KSClassDeclaration
-        className = cd.simpleName.getShortName()
-        packageName = cd.packageName.getShortName()
+        val isAssisted = cd.isAnnotationPresent(AssistedFactory::class)
+        val creatorOrConstructor: KSFunctionDeclaration?
+        val targetClass =
+          if (isAssisted) {
+            val creatorFunction = cd.getAllFunctions().filter { it.isAbstract }.single()
+            creatorOrConstructor = creatorFunction
+            creatorFunction.returnType!!.resolve().declaration as KSClassDeclaration
+          } else {
+            creatorOrConstructor = cd.primaryConstructor
+            cd
+          }
+        className = targetClass.simpleName.getShortName()
+        packageName = targetClass.packageName.getShortName()
```

Comment:
Looks like this line should be `packageName = cd.packageName.asString()`. Right now the factories are built in the wrong folder with the wrong package

Reply:
```suggestion
        packageName = targetClass.packageName.asString()
```